plugin/plugin.go

```diff
@@ -225,13 +225,37 @@ func (p *Plugin) renderValidatorObjectMethod(o *descriptor.DescriptorProto, t st
 	p.P(`}`)
 
 	p.P(`for k, _ := range v {`)
+
 	p.P(`switch k {`)
 	for _, f := range o.GetField() {
 		p.P(`case "`, f.GetName(), `":`)
 		gt := p.getGoType(f.GetTypeName())
 		if p.IsMap(f) {
 			continue
 		}
+		if fExt, err := proto.GetExtension(f.Options, av_opts.E_Field); err == nil && fExt != nil {
+			favOpt := fExt.(*av_opts.AtlasValidateFieldOption)
+			if favOpt.ReadOnly {
+				p.P(`method, _ := ctx.Value("http-method").(string)`)
+				p.P(`if method != "GET" {`)
+				p.P(`return fmt.Errorf("Field %s has readonly access", k)`)
+				p.P(`}`)
+			} else if methods := GetDenyOperations(favOpt.GetDeny()); len(methods) != 0 {
+				p.P(`method, _ := ctx.Value("http-method").(string)`)
+				p.WriteString("if v[k] != nil && (")
+				for ind, m := range methods {
+					p.WriteString(fmt.Sprintf(`method == "%s"`, m))
+					if ind == len(methods)-1 {
+						p.WriteString(") {")
+						break
+					}
+					p.WriteString(" || ")
+				}
+				p.P(`return fmt.Errorf("Field %s has unsupported for method %s", k, method)`)
```

Comment:
simplier is better, maybe just paste

if /* deny.update = true */ {
p.P(`if method == "POST" {`)
p.P(`return fmt.Errorf("Field %q is not allowed for create method??")`)

etc/